openshift_cli_installer/libs/clusters/aro_cluster.py

```diff
@@ -0,0 +1,248 @@
+            self.dump_aro_cluster_data_to_file()
+
+    def dump_aro_cluster_data_to_file(self):
+        self.logger.info(f"Writing {self.platform} cluster {self.cluster_name} data to {self._cluster_data_yaml_file}")
+        with open(self._cluster_data_yaml_file, "r+") as fd:
+            _cluster_data = yaml.safe_load(fd)
+            _cluster_data["cluster"].update(self.cluster)
+            fd.write(yaml.dump(_cluster_data))
+
+    def assert_cluster_params_are_valid(self):
+        def assert_is_supported_version():
+            aro_supported_versions = get_aro_supported_versions(
+                aro_client=self.aro_client,
+                region=self.region,
+            )
+            assert self.version in aro_supported_versions, (
+                f"Version {self.version} is not supported for {self.platform}, supported "
+                f"versions: {aro_supported_versions}"
+            )
+
+        assert_is_supported_version()
+
+    def create_cluster_resources(self):
+        self.create_resource_group()
+        self.create_virtual_network()
+
+    def destroy_cluster_resources(self):
+        self.delete_virtual_network()
+        self.delete_resource_group()
+
+    def create_resource_group(self):
+        self.logger.info(f"Creating resource group {self.resource_group_name}")
+        self.resource_client.resource_groups.create_or_update(
+            resource_group_name=self.resource_group_name, parameters={"location": self.region}
+        )
+
+    def delete_resource_group(self):
+        self.logger.info(f"Deleting resource group {self.resource_group_name}")
+        self.resource_client.resource_groups.begin_delete(resource_group_name=self.resource_group_name)
+
+    def create_virtual_network(
+        self,
+        vnet_address_prefix="10.0.0.0/22",
+        workers_subnet_address_prefix="10.0.0.0/23",
+        master_subnet_address_prefix="10.0.2.0/23",
+    ):
+        def create_subnet(subnet_name, subnet_address_prefix):
+            self.logger.info(f"Creating subnet {subnet_name} in virtual network {self.virtual_network_name}")
+            self.network_client.subnets.begin_create_or_update(
+                resource_group_name=self.resource_group_name,
+                virtual_network_name=self.virtual_network_name,
+                subnet_name=subnet_name,
+                subnet_parameters={"properties": {"addressPrefix": subnet_address_prefix}},
+            ).result()
+
+        self.logger.info(f"Creating virtual network {self.virtual_network_name}")
+        self.network_client.virtual_networks.begin_create_or_update(
+            virtual_network_name=self.virtual_network_name,
+            resource_group_name=self.resource_group_name,
+            parameters={
+                "location": self.region,
+                "address_space": {"address_prefixes": [vnet_address_prefix]},
+            },
+        ).result()
+
+        # Create worker + master nodes subnets
```

Comment:
please remove the comment; code should be explicit

Reply:
done